src/ansys/geometry/core/_grpc/_services/v0/measurement_tools.py

```diff
@@ -0,0 +1,65 @@
+# Copyright (C) 2023 - 2025 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Module containing the measurement tools service implementation for v0."""
+
+import grpc
+
+from ansys.geometry.core.errors import protect_grpc
+
+from ..base.measurement_tools import GRPCMeasurementToolsService
+
+
+class GRPCMeasurementToolsServiceV0(GRPCMeasurementToolsService):
+    """Measurement tools service for gRPC communication with the Geometry server.
+
+    This class provides methods to interact with the Geometry server's
+    measurement tools service. It is specifically designed for the v0 version of the
+    Geometry API.
+
+    Parameters
+    ----------
+    channel : grpc.Channel
+        The gRPC channel to the server.
+    """
+
+    @protect_grpc
+    def __init__(self, channel: grpc.Channel):  # noqa: D102
+        from ansys.api.geometry.v0.measuretools_pb2_grpc import MeasureToolsStub
+
+        self.stub = MeasureToolsStub(channel)
+
+    @protect_grpc
+    def min_distance_between_objects(self, **kwargs):  # noqa: D102
+        from ansys.api.geometry.v0.measuretools_pb2 import MinDistanceBetweenObjectsRequest
+
+        bodies = kwargs["bodies"]
+        selection = kwargs["selection"]
+
+        request = MinDistanceBetweenObjectsRequest(
+            bodies=bodies,
+            selection=selection,
+        )
+
+        response = self.stub.MinDistanceBetweenSelectionObjects(request)
+        return {
+            "distance": response.gap.distance,
+        }
```

Comment:
1. Add the comments on each line ( `` Creation request object, calling stube, returning response...``)
2. Return type hint is missing in the definition (`` -> dict``)
3. Depending on the backend version (<25.2) the request object is different -- this has to be handled here. Make sure to pass the backend version as an input.
4. Remember the GRPC IDs have to be built here -- we should remove the usage of ``._grpc_id`` moving forward